File: 0110-balanced-binary-tree/0110-balanced-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int isBalance(TreeNode* root, int & mx){
        if(root==NULL){
            return 0;
        }
        int lh=isBalance(root->left,mx);
        int rh=isBalance(root->right,mx);
        mx=max(mx,abs(lh-rh));
        return 1+max(lh,rh);
    }
    bool isBalanced(TreeNode* root) {
        int mx=0;
        isBalance(root,mx);
        if(mx>1){
            return false;
        }
        return true;
    }
};

File: 0110-balanced-binary-tree/0110-balanced-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int height(TreeNode root){
        if(root==null){
            return 0;
        }

        int l=height(root.left);
        int r=height(root.right);
        return 1+Math.max(l,r);
    }
    public boolean isBalanced(TreeNode root) {
        if(root==null){
            return true;
        }

        int left=height(root.left);
        int right=height(root.right);

        if(Math.abs(left-right)>1){
            return false;
        }

        return isBalanced(root.left) && isBalanced(root.right);
    }
}